modules/email-service/api/src/main/java/com/innowise/educationalsystem/listener/MailRequestListener.java

```diff
@@ -0,0 +1,48 @@
+package com.innowise.educationalsystem.listener;
+
+import com.innowise.educationalsystem.dto.AttemptRequest;
+import com.innowise.educationalsystem.dto.AttemptResponse;
+import com.innowise.educationalsystem.dto.MailKafkaHeaders;
+import com.innowise.educationalsystem.dto.MailRequest;
+import com.innowise.educationalsystem.dto.MailResponse;
+import com.innowise.educationalsystem.service.AttemptService;
+import com.innowise.educationalsystem.service.MailService;
+import java.util.UUID;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.kafka.annotation.KafkaListener;
+import org.springframework.messaging.handler.annotation.Header;
+import org.springframework.messaging.handler.annotation.Payload;
+import org.springframework.messaging.handler.annotation.SendTo;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class MailRequestListener {
+
+    private final MailService mailService;
+    private final AttemptService attemptService;
+
+    @KafkaListener(topics = "${kafka.topic.mail-request}", containerFactory = "mailRequestKafkaListenerContainerFactory")
+    @SendTo("${kafka.topic.mail-response}")
+    public MailResponse listenMailRequests(
+        @Payload MailRequest mailRequest,
+        @Header(MailKafkaHeaders.KAFKA_MESSAGE_ID) UUID kafkaMessageId,
+        @Header(MailKafkaHeaders.SUBSCRIPTION_ID) UUID subscriptionId) {
+
+        log.info("Received message {} with msgId {} and subId {}", mailRequest, kafkaMessageId, subscriptionId);
+        return mailService.createMail(mailRequest);
```

Comment:
There is a problem here: how does a client know that a particular message is a response to his email request?